### DISCUSSION TITLE
关于 weblate 审核机制

### DISCUSSION THREAD

**Discussion by exzhawk** [CFPAOrg/Minecraft-Mod-Language-Package#893] [General]
### 每条翻译的状态
1.  Untranslated 未翻译
1. Waiting for review 翻译了, 没被review过. **会被当做有效翻译进入实际文件/线上repo**
1. Approved review 批准了. 不能被没有review权限的人修改了.
1. Suggestion 有建议. 所有用户都可以提交建议. 建议的内容不会被当做有效翻译, 不会被提交.

### 用户权限
权限等级一次提高, 排在后面的权限级别拥有前面所有级别的权限.
1. 普通用户. 只能进行建议
1. 有翻译权限的用户. 可以采纳建议, 或直接提交翻译到实际文件/线上repo.
1. 有 review 权限的用户. 可以把条目变成 Approved 状态.

## 当前状态/问题
1. weblate 上的用户都是至少有翻译权限的用户. 可以直接提交翻译到实际文件/线上repo.
1. 有少量用户有 review 权限.
1. 无法阻止不合格翻译直接被提交至线上.
1. 已有的, 以前的翻译内可能有条目有问题.

## Epix 的想法
1. review 所有已有翻译.
1. 把目前有翻译权限的用户降级为只有建议权限的普通用户. 即, 提交的翻译不会进入实际文件/线上repo.
1. [WIP] 有 review 权限的用户保留 review 权限. 将 review 用于标记旧翻译是否已被 review 过

**Comment by TUsama** [comment]
想法的第二条请尽快，因为不知道什么时候又出幺蛾子。

**Reply by GreyElaina** [reply]
我不在的这之前发生了什么...

**Reply by TUsama** [reply]
https://github.com/CFPAOrg/Minecraft-Mod-Language-Package/commit/0af49685242c9e808c9fe7d3e780a329e8f36917

**Comment by baka-gourd** [comment]
这个可以加checkbox吗？我觉得可以（
- [x] 阻止未经审核的条目合并

**Comment by SettingDust** [comment]
- 条目不被 Review 不进入稳定环境，分开发版和稳定版
- 翻译权限用户提供一定数量被 Approved 的键值后，拥有一段时间的 Review 权限
- 用户拥有的权限经过一定数量的有效投诉后移除
